### ISSUE TITLE
npm install cypress no_proxy not working (?)

### ISSUE THREAD

**Issue by pagadalacs** [cypress-io/cypress#8458]
### Current behavior:

 my package.json file : 

    "scripts": {
        ...
    	"preinstall": "npm install --no-package-lock cross-env run-script-os rxjs && npm run install:cypress",
        "install:cypress:linux": "cross-env CYPRESS_INSTALL_BINARY=https://internal.nexus.repo/cypress/4.0.2/cypress-4.0.2-linux-x64.zip npm install cypress@4.0.2 --save-exact",
        ...
      },

npm install  always trying to use http_proxy it seems, and apparently it fails, since its our internal repo doesn't need proxy. i added in no_proxy=.nexus.repo environment variable to bypass proxy.    I have proxy and no_proxy configured in environment variables and also in .npmrc also.  Looks like no_proxy option not working in this case ?     Any help us appreciated ? 

### Desired behavior:

 cypress install should ignore proxy 

### output on jenkins  :
```
npm info lifecycle cypress@4.0.2~postinstall: cypress@4.0.2

> cypress@4.0.2 postinstall /opt/jenkins/workspace/npm_testjob/node_modules/cypress
> node index.js --exec install

⚠ Warning: Forcing a binary version different than the default.

  The CLI expected to install version: 4.0.2

  Instead we will install version: https://internal.nexus.repo/cypress/4.0.2/cypress-4.0.2-linux-x64.zip 

  These versions may not work properly together.

Installing Cypress (version: https://internal.nexus.repo/cypress/4.0.2/cypress-4.0.2-linux-x64.zip )

[?25l[06:10:04]  Downloading Cypress     [started]
[06:10:04]  Downloading Cypress     [failed]
[?25hThe Cypress App could not be downloaded.

Does your workplace require a proxy to be used to access the Internet? If so, you must configure the HTTP_PROXY environment variable before downloading Cypress. Read more: https://on.cypress.io/proxy-configuration

Otherwise, please check network connectivity and try again:
```

**Comment by pagadalacs** [comment]
I solved the issue, its workaround : configured cntlm proxy , that able to bypass proxies for all the internal repos or all domains menioned in "noProxy" . just in case somebody is struggling with same issue as above.

so in env :

`export http_proxy = http:/10.0.0.10:3128 ( cntlm proxy)`
in cntlm proxy :

```
 Proxy           http://corporateproxy     
 NoProxy         localhost, 127.0.0.*, .nexus.repo 
```

**Comment by l3ender** [comment]
We are facing the same issue in our build environment (Jenkins), but using cntlm is not an option in our environment.

Environment config is as follows (proxy configured for other builds):
```
HTTP_PROXY=http://webproxy.myorg.local:9091
HTTPS_PROXY=http://webproxy.myorg.local:9091
CYPRESS_INSTALL_BINARY=http://nexus.myorg.local:8081/repository/build-tools/linux/cypress.zip
NO_PROXY=nexus.myorg.local
```

Yet we still see the following error, which indicates the traffic still going to the proxy servers:
```
> cypress@5.6.0 postinstall /opt/build-tools/jenkins-work-dir/workspace/new-member.test/node_modules/cypress
> node index.js --exec install

? Warning: Forcing a binary version different than the default.

  The CLI expected to install version: 5.6.0

  Instead we will install version: http://nexus.myorg.local:8081/repository/build-tools/linux/cypress.zip

  These versions may not work properly together.

Installing Cypress (version: http://nexus.myorg.local:8081/repository/build-tools/linux/cypress.zip)

[?25l[16:38:35]  Downloading Cypress     [started]
[16:38:35]  Downloading Cypress     [failed]
[?25hThe Cypress App could not be downloaded.

Does your workplace require a proxy to be used to access the Internet? If so, you must configure the HTTP_PROXY environment variable before downloading Cypress. Read more: https://on.cypress.io/proxy-configuration

Otherwise, please check network connectivity and try again:

----------

URL: http://nexus.myorg.local:8081/repository/build-tools/linux/cypress.zip
Error: Failed downloading the Cypress binary.
Response code: 407
Response message: Proxy Authentication Required
```

Any suggestions?

**Comment by geekrumper** [comment]
Any remarks on this one, is it maybe solved in a version after 6.2.1? We are facing the same issue at my company.

We need proxy for our fonts at `fonts.here.net` so the page can be fully loaded while cypress is being executed and therefore set proxy in jenkins

```
env.HTTP_PROXY = "http://getout.here.net:3128/"
env.HTTPS_PROXY = "http://getout.here.net:3128/"
env.NO_PROXY = "127.0.0.1,localhost,.here.net,.there.net,gitit.here.net,.tools.here.net"
```
as you can see, we try to keep everything internally as possible.

We provide the install process of npm with the following param:

`CYPRESS_INSTALL_BINARY = https://artifactory.tools.here.net/artifactory/ext-binaries-local/io/cypress/6.2.1/linux/x64/cypress.zip`

In short `fonts.here.net` should be able to go out and `artifactory.tools.here.net` should fall under the no_proxy rule, which is not the case.

**Comment by l3ender** [comment]
@jennifer-shehane would you mind looking at this issue?

**Comment by flotwig** [comment]
Since the OP was able to resolve ther issue with configuration, I suggest checking their approach and seeing if it works. If you are still encountering problems, please open a new issue.
